File: v131016/TeeNew/Polar_Smooth.pas
unit Polar_Smooth;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QExtCtrls, QStdCtrls, QComCtrls,
  QImgList,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls, ComCtrls,
  ImgList,
  {$ENDIF}
  Base, TeeProcs, TeEngine, Chart, Series, TeePolar, TeeSpline;

type
  TPolarSmoothForm = class(TBaseForm)
    Series2: TPolarSeries;
    TeeFunction1: TSmoothingFunction;
    Series1: TPolarSeries;
    CheckBox1: TCheckBox;
    CheckBox2: TCheckBox;
    GroupBox1: TGroupBox;
    CheckBox3: TCheckBox;
    procedure CheckBox3Click(Sender: TObject);
    procedure CheckBox1Click(Sender: TObject);
    procedure CheckBox2Click(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.DFM}
{$ELSE}
{$R *.xfm}
{$ENDIF}

procedure TPolarSmoothForm.CheckBox3Click(Sender: TObject);
begin
  TeeFunction1.Interpolate:= CheckBox3.Checked;
end;

procedure TPolarSmoothForm.CheckBox1Click(Sender: TObject);
begin
  Series1.Visible:= CheckBox1.Checked;
end;

procedure TPolarSmoothForm.CheckBox2Click(Sender: TObject);
begin
  Series2.Visible:= CheckBox2.Checked;
end;

initialization
  RegisterClass(TPolarSmoothForm);
end.

File: TeeNew/Builder2007/Polar_Smooth.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Polar_Smooth.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma link "TeePolar"
#pragma link "TeeSpline"
#pragma resource "*.dfm"
TPolarSmoothForm *PolarSmoothForm;
//---------------------------------------------------------------------------
__fastcall TPolarSmoothForm::TPolarSmoothForm(TComponent* Owner)
  : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TPolarSmoothForm::CheckBox3Click(TObject *Sender)
{
  TeeFunction1->Interpolate = CheckBox3->Checked;
}
//---------------------------------------------------------------------------

void __fastcall TPolarSmoothForm::CheckBox1Click(TObject *Sender)
{
  Series1->Visible = CheckBox1->Checked;
}
//---------------------------------------------------------------------------

void __fastcall TPolarSmoothForm::CheckBox2Click(TObject *Sender)
{
  Series2->Visible = CheckBox2->Checked;
}
//---------------------------------------------------------------------------

